Packs/FidelisEndpoint/Integrations/FidelisEndpoint/FidelisEndpoint.py

```diff
@@ -0,0 +1,1710 @@
+KILL_PROCESS_WINDOWS = '8d379688-dde1-451d-8fa2-4f29c84baf97'  # disable-secrets-detection
+KILL_PROCESS_MAC_LINUX = '76577d3a-c1d7-4d10-af9e-5825c3f9d016'  # disable-secrets-detection
+
+# Delete file
+DELETE_FILE_WINDOWS = '11cb4fae-5516-4391-8a3c-eb09793cd5dd'  # disable-secrets-detection
+DELETE_FILE_MAC_LINUX = 'bead9799-401d-4b9e-adca-cf41b20c9118'  # disable-secrets-detection
+
+# Network isolation
+NETWORK_ISOLATION_WINDOWS = '1d01cc84-753d-4060-89a7-463567552a62'  # disable-secrets-detection
+NETWORK_ISOLATION_MAC_LINUX = 'fd09996a-ef56-49fb-b811-0e5da4bd07ca'  # disable-secrets-detection
+
+# Remove network isolation
+REMOVE_NETWORK_ISOLATION_WINDOWS = '99bbaea5-df18-40cc-8759-b5fb61527d5a'  # disable-secrets-detection
+REMOVE_NETWORK_ISOLATION_MAC_LINUX = '5e252298-4c50-4cdd-94c0-d6997b79157c'  # disable-secrets-detection
+
+
+class Client(BaseClient):
+    """
+    Client to use in the Fidelis Endpoint integration. Overrides BaseClient
+    """
+    def __init__(self, server_url: str, username: str, password: str, verify: bool, proxy: bool, headers: dict):
+
+        super().__init__(base_url=server_url, verify=verify, proxy=proxy, headers=headers)
+        self._username = username
+        self._password = password
+        self._token = self._generate_token()
+
+        if self._token:
+            token = self._token
+            headers.update({'Authorization': 'bearer ' + token})
+
+    def _generate_token(self) -> str:
+        """Generate a token
+
+        Returns:
+            token valid for 10 minutes
+        """
+        params = {
+            'username': self._username,
+            'password': self._password
+        }
+        response = self._http_request('GET', '/authenticate', params=params)
+        if response.get('error'):
+            raise Exception(response.get('error'))
+        token = response.get('data', {}).get('token', '')
+
+        return token
+
+    def test_module_request(self):
+        """Performs basic GET request to check if the API is reachable and authentication is successful.
+
+        Returns:
+            Response content
+        """
+        suffix = '/alerts/getalertsV2'
+        self._http_request('GET', suffix, params={'take': 1})
+
+    def list_alerts(self, limit: str = None, sort: str = None, start_date=None, end_date=None) -> Dict:
+
+        url_suffix = '/alerts/getalertsV2'
+        params = assign_params(
+            take=limit,
+            sort=sort,
+            startDate=start_date,
+            endDate=end_date
+        )
+
+        return self._http_request('GET', url_suffix, params=params)
+
+    def get_host_info(self, host_name: Union[None, str], ip_address: Union[None, str]):
+
+        if host_name:
+            url_suffix = f'/endpoints/v2/0/100/hostname%20Ascending?accessType=3&search={{%22searchFields%22: \
+                             [{{%22fieldName%22:%22HostName%22,%22values%22:[{{%22value%22:%22{host_name}%22}}]}}]}}'
+
+        if ip_address:
```

Comment:
in all cases you need either `host_name` or `ip_address`, please change to `elif`